src/dbnode/storage/shard.go

```diff
@@ -2372,6 +2377,32 @@ func (s *dbShard) Snapshot(
 	}
 	s.RUnlock()
 
+	var (
+		needsSnapshot bool
+		// NB(bodu): This always defaults to false if the record does not exist.
+		emptySnapshotOnDisk = s.emptySnapshotOnDiskByTime[blockStart]
```

Comment:
Seems like we should hold the RLock while reading this data?

Reply:
Snapshotting happens in a single thread so all reads/writes also happens in that thread so holding a lock is not necessary I think?

Reply:
I have a comment on on the field definition:
```
// NB(bodu): Cache state on whether we snapshotted last or not to avoid
// going to disk to see if filesets are empty. We don't need to lock on this
// because snapshots happen sequentially.
emptySnapshotOnDiskByTime map[time.Time]bool
```